rfc/0002/README.md

```diff
@@ -0,0 +1,198 @@
+Rabbit storage for containerized applications
+=============================================
+
+**authors**: Nate Roiger <nate.roiger@hpe.com> |
+**state**: ideation
+
+---
+
+For Rabbit to provide storage to a containerized application there needs to be _some_ mechanism. The remainder of this RFC proposes that mechanism.
+
+Actors
+------
+
+There are several different actors involved
+
+- The AUTHOR of the containerized application
+- The ADMINISTRATOR who works with the author to determine the application requirements for execution
+- The USER who intends to to use the application using the 'container' directive in their job specification
+- The RABBIT software that interprets the #DWs and starts the container during execution of the job
+
+There are multiple relationships between the actors:
+- AUTHOR to ADMINISTRATOR: The author tells the administrator how their application is executed and the NNF storage requirements.
+- Between the AUTHOR and USER: The application expects certain storage, and the #DW must meet those expectations.
+- ADMINISTRATOR to RABBIT: Admin tells Rabbit how to run the containerized application with the required storage.
+- Between USER and RABBIT: User provides the #DW container directive in the job specification. Rabbit validates and interprets the directive.
+
+Proposal
+--------
+
+The proposal below might take a couple of read-throughs; I've also added a concrete example afterward that might help.
+
+1. The AUTHOR writes their application expecting NNF Storage at specific locations. For each storage requirement, they define:
+    1. a unique name for the storage which can be referenced in the 'container' directive
+    2. the expected storage types; if necessary
+    3. the required mount path or mount path prefix
+    4. other constraints or storage requirements (e.g. minimum capacity)
+2. The AUTHOR works with the ADMINISTRATOR to define:
+    1. a unique name for the program to be referred by USER
+    2. the pod template specification for executing their program
+    3. the NNF storage requirements described above. 
+3. The ADMINISTRATOR creates a corresponding _NNF Container Profile_ custom kubernetes resource with the necessary NNF storage requirements and pod specification as described by the AUTHOR
+4. The USER who desires to use the application works with the AUTHOR and the related NNF Container Profile to understand the storage requirements.
+5. The USER submits a WLM job with the #DW container fields populated
+6. WLM runs the job and drives the job through the following stages...
+    1. Proposal: RABBIT validates the #DW container directive by comparing the supplied values to what is listed in the NNF Container Profile. If the USER fails to meet the requirements, the job fails. 
+    2. Pre-run: RABBIT software will:
```

Comment:
Will they be able to indicate whether the container should be pre-run or post-run?

Reply:
Is that a requirement?

Reply:
For some reason I thought the container would be a post-run thing.

Reply:
My understanding that the container would be a part of the entire life cycle...  In the case of a new or experimental file system, I think that it would have to be there for setup as well as the rest of the states.

Reply:
This would complicate the workflow since we don't allocate storage until setup. If this is a requirement then we can pursue it, but for now I'd like it to remain in pre-run.